File: my-code/9/main.py
# -*- coding: utf-8 - *-
# @Time     : 2021/05/28 17: 49
# @Author   : Ranshi
# @File     : 9. 回文数.py


class Solution:
    def isPalindrome(self, x: int) -> bool:
        str_x = str(x)
        re_x = str_x[::-1]
        if str_x == re_x:
            return True
        return False


if __name__ == "__main__":
    print(Solution().isPalindrome(123321))

File: my-code/9/main.js
// @Time     : 2021/05/28 17: 47
// @Author   : Ranshi
// @File     : 9. 回文数.js

const isPalindrome = function (x) {
  if (x < 0 || (x % 10 == 0 && x != 0)) {
    return false
  }
  let idx = 0
  while (x > idx) {
    idx = idx * 10 + x % 10
    x = Math.floor(x / 10)
  }
  return x === idx || x == Math.floor(idx / 10)
}


console.log(isPalindrome(x = 121))